api/institutions/serializers.py

```diff
@@ -201,13 +201,38 @@ def get_absolute_url(self, obj):
             },
         )
 
+class UniqueDeptIDField(IDField):
+    """Creates a unique department ID of the form "<institution-id>-<dept-id>"."""
+
+    def __init__(self, *args, **kwargs):
+        kwargs['source'] = kwargs.pop('source', 'name')
+        kwargs['help_text'] = kwargs.get('help_text', 'Unique ID that is a compound of two objects. Has the form "<institution-id>-<dept-id>". Example: "cos-psych"')
+        super(UniqueDeptIDField, self).__init__(*args, **kwargs)
+
+    def _get_resource_id(self):
+        return self.context['request'].parser_context['kwargs']['institution_id']
```

Comment:
remember you can always `self.whatever = self.context['request'].parser_context['kwargs']['whatever_id']` in the `__init__`. I'm 90% sure.